src/AutoUI/Filters/PersistentFilters.tsx

```diff
@@ -229,23 +225,26 @@ export const PersistentFilters = ({
 				});
 			}
 		},
-		[window.location.pathname, onFiltersChange],
+		[onFiltersChange, analytics.webTracker, history, locationSearch],
 	);
 
-	// When the component mounts, filters from the page URL,
-	// then communicate them back to the parent component.
 	React.useEffect(() => {
-		// Make sure we only call onFiltersUpdate on mount once, even if
-		// we are rendering each part of the Filter component separately.
+		if (hasMounted.current) {
+			return;
+		}
+		hasMounted.current = true;
+
+		// Normalize renderMode to handle both array and single value
 		const normalizedRenderMode = new Set(
 			Array.isArray(otherProps.renderMode)
 				? otherProps.renderMode
 				: [otherProps.renderMode],
 		);
+
 		if (normalizedRenderMode.has('all') || normalizedRenderMode.has('add')) {
 			onFiltersUpdate?.(storedFilters);
 		}
-	}, []);
+	}, [onFiltersUpdate, storedFilters, otherProps.renderMode]);
```

Comment:
So `hasMounted` was only added for make this work only on mount? What I've done in the UI in these cases is just comment eslint disable next line for exhaustive deps and include in that comment that we only want this hook to run on mount

Is there a reason using `hasMounted` would be better than acknowledging that we only need this to run on mount via a comment like I mentioned? Disadvantage-wise, this hook just keeps getting re-triggered for no reason whenver storedFilters update 😕